hack/install-golangci-lint.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+VERSION="${1:-1.21.0}"
+
+[ $(golangci-lint version 2>&1 | grep -ci "${VERSION}") -eq 1 ] && exit 0
+curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b $(go env GOPATH)/bin "${VERSION}"
```

Comment:
what does this script do and why is it better than another form of installing it?

Reply:
golangci-lint print version into stderr , so use grep to check version. 
when golangci-lint version is  what we want ,it will do nothing and exit 0. otherwise use curl to download .

another install method is use `go get` . it will change some lib version, i think check tool should not change the src librarys, so i ues curl to download golangci-lint by version

Hi , @cblecker  I'm not comfortable with this install method too, but I'm not sure it is correct to use `go get`